docs/explanations/lifecycle.md

```diff
@@ -2,43 +2,44 @@
 
 The following demonstrates exactly what the code does with a plan through its lifecycle 
 of being written, loaded and run. Take the following plan.
-```
-    from typing import Any, List, Mapping, Optional, Union
-    
-    import bluesky.plans as bp
-    from bluesky.utils import MsgGenerator
-    from dodal.common import inject
-    from bluesky.protocols import Readable
-    
-    
-    def count(
-        detectors: List[Readable] = [inject("det")],  # default valid for Blueapi only
-        num: int = 1,
-        delay: Optional[Union[float, List[float]]] = None,
-        metadata: Optional[Mapping[str, Any]] = None,
-    ) -> MsgGenerator:
-        """
-        Take `n` readings from a collection of detectors
-    
-        Args:
-            detectors (List[Readable]): Readable devices to read: when being run in Blueapi
-                                        defaults to fetching a device named "det" from its
-                                        context, else will require to be overridden.
-            num (int, optional): Number of readings to take. Defaults to 1.
-            delay (Optional[Union[float, List[float]]], optional): Delay between readings.
-                                                                Defaults to None.
-            metadata (Optional[Mapping[str, Any]], optional): Key-value metadata to include
-                                                            in exported data.
+
+```python
+import bluesky.plans as bp
+
+from typing import Any, List, Mapping, Optional, Union
+
+from blueapi.core import MsgGenerator
+from bluesky.protocols import Readable
```

Comment:
Should:

```suggestion
from bluesky.protocols import Readable
from bluesky.utils import MsgGenerator
```

It would be a nit, but we've seen how much of an issue this line has caused!!